File: Tema_2/Ejercicio_1234/app/src/main/java/com/example/ejercicio_1234/MainActivity.java
package com.example.ejercicio_1234;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    int n=0;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        n=10;
        if(savedInstanceState!=null){
            String nombre= (String) savedInstanceState.get("nombre");
            Toast.makeText(this, nombre, Toast.LENGTH_LONG).show();

        }
    }

    protected void onSaveInstanceState(Bundle saveInstanceState) {
        Toast.makeText(this, "Método saveInstanceState "+n, Toast.LENGTH_LONG).show();
        saveInstanceState.putString("nombre","Enrique");
        super.onSaveInstanceState(saveInstanceState);
    }

    protected void onStart() {
        super.onStart();

        Toast.makeText(this, "Método onstart "+n, Toast.LENGTH_LONG).show();
    }
    protected void onResume() {
        super.onResume();
        Toast.makeText(this, "Método onResume "+n, Toast.LENGTH_LONG).show();
    }
    protected void onPause() {
        super.onPause();
        Toast.makeText(this, "Método onPause "+n, Toast.LENGTH_LONG).show();
    }
    protected void onStop() {
        super.onStop();
        n=5;
        Toast.makeText(this, "Método onStop "+n, Toast.LENGTH_LONG).show();
    }
    protected void onDestroy() {
        super.onDestroy();
        Toast.makeText(this, "Método onDestroy "+ n, Toast.LENGTH_LONG).show();
    }

}

File: ADA_Tema_6/KotlinBaseDatosExamen/app/src/main/java/com/example/kotlinbasedatosexamen/MainActivity.kt
package com.example.kotlinbasedatosexamen

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.example.kotlinbasedatosexamen.room.AppDB
import com.example.kotlinbasedatosexamen.room.entities.Alimento
import com.example.kotlinbasedatosexamen.room.entities.Ingrediente
import com.example.kotlinbasedatosexamen.room.entities.Receta

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var alimento= Alimento(nombre = "Filete de pollo", proteinas = 150, grasas = 75, hidratos = 30)
        AppDB.getAppDB(applicationContext)!!.daoAlimento().crearAlimento(alimento)

        val alimentos: List<Alimento>? =
            AppDB.getAppDB(applicationContext)!!.daoAlimento().verAlimentos()
            Log.d(":::Recetas", "Lectura de alimentos")
        if (alimentos != null) {
            for (al in alimentos) {
                Log.d(":::Recetas",
                    al.mostrar())
            }
        }

        var receta= Receta(nombre = "Filete de pollo cocinado")
        AppDB.getAppDB(applicationContext)!!.daoReceta().crearReceta(receta)

        val recetas: List<Receta>? =
            AppDB.getAppDB(applicationContext)!!.daoReceta().verRecetas()
        Log.d(":::Recetas", "Lectura de recetas")
        if (recetas != null) {
            for (re in recetas) {
                Log.d(":::Recetas",
                    re.mostrar())
            }
        }
        var ingrediente=Ingrediente(1,1,1)
        AppDB.getAppDB(applicationContext)!!.daoIngrediente().crearIngrediente(ingrediente)

        val ingredientes: List<Ingrediente>?=
            AppDB.getAppDB(applicationContext)!!.daoIngrediente().verIngredientes()
        Log.d(":::Recetas","Lectura de ingredientes")
        if (ingredientes != null){
            for (ing in ingredientes){
                var recetaIng=AppDB.getAppDB(applicationContext)!!.daoReceta().verReceta(ing.id_receta!!)
                Log.d(":::Recetas",ing.mostrar(applicationContext))
            }
        }

        AppDB.getAppDB(applicationContext)!!.daoIngrediente().borrarIngrediente(AppDB.getAppDB(applicationContext)!!.daoIngrediente().verIngrediente(1))
        AppDB.getAppDB(applicationContext)!!.daoAlimento().borrarAlimento(AppDB.getAppDB(applicationContext)!!.daoAlimento().verAlimento(1))
        AppDB.getAppDB(applicationContext)!!.daoReceta().borrarReceta(AppDB.getAppDB(applicationContext)!!.daoReceta().verReceta(1))
    }
}